src/com/epam/javatrainings/annotations/validator/Validator.java

```diff
@@ -0,0 +1,60 @@
+package com.epam.javatrainings.annotations.validator;
+
+import com.epam.javatrainings.annotations.custom_annotations.*;
+
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validator<T>{
+
+    Pattern VALID_EMAIL_ADDRESS_REGEX;
+    Matcher matcher;
+    List<String> errorList = new ArrayList<>();
+
+    public List<String> validate(T dto) throws IllegalAccessException, NoSuchFieldException {
+        Field discountRate = dto.getClass().getDeclaredField("discountRate");
+        if (discountRate.isAnnotationPresent(Min.class) && discountRate.isAnnotationPresent(Max.class)) {
+            discountRate.setAccessible(true);
+            Min min = discountRate.getAnnotation(Min.class);
+            Max max = discountRate.getAnnotation(Max.class);
+            if ((int) discountRate.get(dto) < min.min() || max.max() < (int) discountRate.get(dto)) {
+                errorList.add("Discount value is wrong!!");
+            }
+        }
+        Field name = dto.getClass().getDeclaredField("name");
+        if (name.isAnnotationPresent(Length.class)) {
```

Comment:
you can move logic in if statement into a separate class, and think to make it more robust and generic, what if lenght annotation is placed on int field?